CrisisTextLine/Elmer
# add individual deletion of shifts

#### What's this PR do?
1. adds the individual deletion of shifts functionality to Elmer. 
2. Also removes the bulk deletion route of shifts. 

#### What's the user flow? 
1. When the user navigates to online.crisistextline.org/schedules, she'll see a new button to the page which says "drop individual shifts" (to be added.) 
2. Clicking this button will GET the `/scheduling` route on Elmer, displaying a page which lists all the user's shifts in a checkbox form. Only two checkboxes are able to be checked at one time, roughly reflecting WhenIWork's API limit. 
3. When the user checks a shift's box and submits the form, we POST to the `/delete-shifts` route. 
4. The user is redirected to a new view: `someShiftsCancelled.jade`, which then gives a link for the user to be redirected back to When I Work, so that they can schedule another shift. 

#### How should this be manually tested?
Note below that the app crashes upon a successful delete shift request, so we can't confirm that the final page is displayed. 

Currently has been tested by scheduling two shifts in the `test` location on WiW, then deleting them together or individually with Elmer after commenting out the checks for a valid email and token. 
#### Background:
**note 1**
**TODO**: Currently, our successful delete-shift batch requests consistently return with an 502 error encoded in HTML in the body of the response. But this breaks line 159 within `index.js` of the module `wheniwwork_unofficial`. We need to resolve this--a crappy solution that might be the best solution would be to add a check for HTML on line 159. 

Currently, this crashing breaks the redirect after shifts are cancelled. 

**note 2**
Since the shift recurrence cronjob currently runs every minute, it's possible that a user books shifts and attempts to delete them before the shift recurrence cronjob has successfully recurred that shift. 

If this happens, the shifts won't delete--much of the shift deletion logic relies on the `notes.parent_shift` param, which is only added during shift recurrence--and instead the user will be redirected to an error message telling her to wait 30 seconds and then try again.
